### PR TITLE
M3 install bundle

### PR DESCRIPTION
**Pull request by hluchas**
**Please be sure you are submitting this against the _staging_ branch.**

[//]: # ( Please answer the following questions: )

| Q  | A
| --- | ---
| Bug fix? | 
| New feature? | 
| Automated tests included? |
| Related user documentation PR URL | 
| Related developer documentation PR URL | 
| Issues addressed (#s or URLs) | https://github.com/mautic/mautic/issues/7991
| BC breaks? | Y
| Deprecations? | 

[//]: # ( Note that all new features should have a related user and/or developer documentation PR in their respective repositories. )

[//]: # ( Required: )
#### Description:
Installation bundle upgrade

#### Steps to test this PR:
1. Delete local.php in config dir
2. Try to install and reinstall with table backup to the same DB

#### List backwards compatibility breaks:
1. Removed unused \Mautic\InstallBundle\Configurator\Step\CheckStep::$install_source
2. Remove unused \Mautic\InstallBundle\Configurator\Step\DoctrineStep::$server_version

Based on this PR https://github.com/mautic/mautic/pull/8068

### CODE REVIEW

**Comment by hluchas** [comment]
@escopecz This might be an issue
Snímek obrazovky 2019-12-13 v 12 12 04

File: app/bundles/InstallBundle/Configurator/Step/CheckStep.php [hunk 1]
```diff
@@ -136,8 +128,12 @@ public function checkRequirements()
...
         }
 
-        if (get_magic_quotes_gpc()) {
-            $messages[] = 'mautic.install.magic_quotes_enabled';
+        if (
+            version_compare(PHP_VERSION, '5.6.0', '>=')
+            &&
+            version_compare(PHP_VERSION, '7', '<')
+            ) {
+            $messages[] = 'mautic.install.always_populate_raw_post_data_enabled';
```
Comment (escopecz) [inline app/bundles/InstallBundle/Configurator/Step/CheckStep.php]: This is not necessary as M3 will run on PHP 7.2+

File: app/bundles/InstallBundle/Configurator/Step/EmailStep.php [hunk 2]
```diff
@@ -1,10 +1,10 @@
 <?php
 
-/*
+/**
```
Comment (escopecz) [inline app/bundles/InstallBundle/Configurator/Step/EmailStep.php]:
```suggestion
/*
```

File: app/bundles/InstallBundle/Configurator/Step/CheckStep.php [hunk 3]
```diff
@@ -36,18 +33,22 @@ class CheckStep implements StepInterface
...
 
-    /** @var OpenSSLCipher */
+    /**
+     * @var OpenSSLCipher
+     */
     private $openSSLCipher;
 
     /**
      * Absolute path to cache directory.
+     * Required in step.
```
Comment (escopecz) [inline app/bundles/InstallBundle/Configurator/Step/CheckStep.php]: What this comment means?

File: app/bundles/InstallBundle/Configurator/Form/EmailStepType.php [hunk 4]
```diff
@@ -167,20 +169,13 @@ public function buildForm(FormBuilderInterface $builder, array $options)
...
-                        'mautic.email.config.mailer_auth_mode.plain',
-                        'mautic.email.config.mailer_auth_mode.login',
-                        'mautic.email.config.mailer_auth_mode.cram-md5',
-                    ]
-                ),
+                'choices' => [
+                    'plain'    => 'mautic.email.config.mailer_auth_mode.plain',
+                    'login'    => 'mautic.email.config.mailer_auth_mode.login',
+                    'cram-md5' => 'mautic.email.config.mailer_auth_mode.cram-md5',
+                ],
```
Comment (escopecz) [inline app/bundles/InstallBundle/Configurator/Form/EmailStepType.php]: Note: We will have to test these selects carefully. This refactoring out of ChoiceList created bugs in the Page Builder.

File: app/bundles/InstallBundle/Configurator/Step/CheckStep.php (hunk 3 above)
Comment (hluchas) [reply app/bundles/InstallBundle/Configurator/Step/CheckStep.php]: Usage cannot be found in PHPStorm code inspection. This caused problems form me whith usages autocheck and is covered by some magic in the top of these steps.

That's why I leaved notice about this.

File: app/bundles/InstallBundle/Configurator/Form/EmailStepType.php (hunk 4 above)
Comment (hluchas) [reply app/bundles/InstallBundle/Configurator/Form/EmailStepType.php]: Tested again and works well

File: app/bundles/InstallBundle/Configurator/Step/CheckStep.php (hunk 3 above)
Comment (hluchas) [reply app/bundles/InstallBundle/Configurator/Step/CheckStep.php]: Btw. I removed some attrs based on this check.

**Comment by escopecz** [comment]
@hluchas there are tests failing in m2-to-m3 branch. This PR should fix them: https://github.com/mautic/mautic/pull/8232

Please rebase once that PR is merged.

**Comment by dongilbert** [comment]
@escopecz Will these failing tests be resolved by #8232

**Comment by escopecz** [comment]
Haha, same time :) Yes, they should.

**Comment by escopecz** [comment]
@hluchas please rebase or merge m2-to-m3 branch. I'd do it but doesn't seem to have permission.

**Comment by escopecz** [comment]
I take it back. It was enough to re-run the CI job. Tests are passing. Will merge once CS fixer finishes.

**Pull request merged**